### PR TITLE
Ensure that we only reduce extension methods from the same language.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/Roslyn/issues/8036

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by jasonmalinowski** [comment]
Tests?

File: src/Features/CSharp/Portable/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -648,7 +648,8 @@ protected override bool IsViableExtensionMethod(IMethodSymbol method, SyntaxNode
             var semanticInfo = semanticModel.GetTypeInfo(leftExpression, cancellationToken);
             var leftExpressionType = semanticInfo.Type;
 
-            return leftExpressionType != null && method.ReduceExtensionMethod(leftExpressionType) != null;
+            IMethodSymbol reducedMethod;
+            return semanticModel.Compilation.TryReduceExtensionMethod(method, leftExpressionType, cancellationToken, out reducedMethod);
```
Comment (jasonmalinowski) [inline src/Features/CSharp/Portable/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: Can this be refactored into the common base somehow?

File: src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs [hunk 2]
```diff
@@ -133,5 +135,32 @@ public static INamedTypeSymbol SuppressMessageAttributeType(this Compilation com
...
+                receiver = SymbolFinder.FindSimilarSymbols(receiver, compilation, cancellationToken).FirstOrDefault();
+            }
+
+            if (method == null || receiver == null)
+            {
+                reducedMethod = null;
+                return false;
+            }
+
+            reducedMethod = method.ReduceExtensionMethod(receiver);
```
Comment (jmarolf) [inline src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs]: are we guaranteed that FindSimilarSymbols always returns a symbol that matches the compilation language or null?

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs]: FindSimilarSymbols always returns symbols from the passed compilation.

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs]: FindSimilarSymbols returns symbols from the passed in compilation.  LIterally from inside it.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski feedback incorporated.   Plz review when you get a chance.

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by DustinCampbell** [comment]
:+1:

**Pull request merged**